visitor/typescript/book-export-visitor.interface.ts

```diff
@@ -0,0 +1,8 @@
+import {Book} from "./book";
+
+// Her ConcreteVisitor örneği için metot temsil eder.
+// Parametre olarak Element sınıfından türeyen sınıfların örneğini alır.
+// UML diyagramındaki Visitor yapısına denk gelir.
+export interface IBookExportVisitor {
+  exportVisit(magazine: Book): void;
```

Comment:
```typescript

import {Book} from "./book";

// Her ConcreteVisitor örneği için metot temsil eder.
// Parametre olarak Element sınıfından türeyen sınıfların örneğini alır.
// UML diyagramındaki Visitor yapısına denk gelir.
export interface IBookExportVisitor {
  exportVisit(magazine: Book): void;
  exportVisit(encyclopedia: Encyclopedia): void;
}
```

Burada sanki 2 adet metot olacak gibi, gözden kaçmış olabilir mi acaba?  Diğer bunu implement eden sınıflarda da uygulanması gerekiyor. Göz attım da C#, Java vs. oradaki uygulamaları doğru ama Python kısmında farklı işlenmiş, oradan baktınız herhalde. 🤔

Reply:
Temel anlamda bakınca `Encyclopedia` ve `Magazine` de bir `Book` bundan dolayı sizin uygulama da doğru. Bir yorumunu rica edeceğim.

Reply:
*(`exportVisit(magazine: Book): void` kısmında magazine gözden kaçmış, book olması gerekiyor)*

Biraz dil kısıtlaması yüzünden, method overloading C# ve Java'nın aksine TypeScript tarafında şu şekilde;
![Screen Shot 2022-05-16 at 21 16 45](https://user-images.githubusercontent.com/24798543/168656887-a9515010-1419-4e40-b0d5-8181e9ea5f8b.png)

Ben de method overloading yerine `Encyclopedia` ve `Magazine`'in türetildiği `Book` üzerinden tanımlamıştım, güncellemeyi push'luyorum 👍